File: src-p/infra/adapters/proxy/BlockEngine.py
import re
from domain.proxy.IBlockEngine import IBlockEngine
from domain.proxy.ProxyRules import MergedRules


class BlockEngine(IBlockEngine):
    def is_blocked(self, text: str, rules: MergedRules) -> bool:
        for sub in rules.blockUrls:
            if sub in text:
                return True
        for pat in rules.blockPatterns:
            if re.search(pat, text, re.IGNORECASE):
                return True
        return False

File: src/infra/adapters/proxy/reverse/BlockEngine.ts
import { IBlockEngine } from "../../../../domain/proxy/reverse/IBlockEngine.js";
import { MergedRules } from "../../../../domain/proxy/reverse/ProxyRules.js";

export class BlockEngine implements IBlockEngine {
  isBlocked(text: string, rules: MergedRules): boolean {
    for (const sub of rules.blockUrls) {
      if (text.includes(sub)) return true;
    }
    for (const pat of rules.blockPatterns) {
      if (new RegExp(pat, "i").test(text)) return true;
    }
    return false;
  }
}

